admin-dev/themes/new-theme/js/nav_bar.js

```diff
@@ -24,24 +24,26 @@
  */
 
 import PerfectScrollbar from 'perfect-scrollbar';
-import '../node_modules/perfect-scrollbar/css/perfect-scrollbar.css';
+import 'perfect-scrollbar/css/perfect-scrollbar.css';
+import getAnimationEvent from './app/utils/animations';
 
 const $ = window.$;
 
 export default class NavBar {
   constructor() {
-
     $(() => {
       const $navBar = $('.nav-bar');
       new PerfectScrollbar($navBar.get(0));
+      let endTransitionEvent = getAnimationEvent('transition', 'end');
+      let transitionFired = false;
```

Comment:
I think it would be best to have a stateful object that handles this all on its own. For example:

```js
class NavbarTransitionHandler {
  constructor($navBar, $mainMenu, endTransitionEvent) {
    this.transitionFired = false;
    this.$navBar = $navBar[0];
    this.$mainMenu = $mainMenu;
    this.endTransitionEvent = endTransitionEvent;
  }

  toggle() {
    if (!this.transitionFired) {
      transitionFired = true;
      this.$navBar.addEventListener(this.endTransitionEvent, this.showNavBarContent);
    } else {
      this.$navBar.removeEventListener(this.endTransitionEvent, this.showNavBarContent);
      transitionFired = false;
    }
  }

  _showNavBarContent(event) {
    if (event.propertyName == 'width') {
      this.$navBar.removeEventListener(endTransitionEvent, showNavBarContent);

      this.$mainMenu.toggleClass('sidebar-closed');

      this.transitionFired = false;
    }
  }
}
```

Then:

```js
$(() => {
  const $navBar = $('.nav-bar');
  new PerfectScrollbar($navBar.get(0));
  let navbarTransitionHandler = new NavbarTransitionHandler(
     $navBar,
     $('.main-menu'),
     getAnimationEvent('transition', 'end');
  );

  // ... snip

  $navBar.on(
    'click',
    '.menu-collapse',
    function onNavBarClick() {
      $('body').toggleClass('page-sidebar-closed');
 
      navbarTransitionHandler.toggle();
```

This yields more loosely-coupled code that's easier to understand.